core/views.py

```diff
@@ -0,0 +1,150 @@
+"""
+Task Manager Core Views
+"""
+from django.utils.encoding import force_text
+from django.utils.http import urlsafe_base64_decode
+from requests import Response
+from rest_framework import status
+from rest_framework.generics import (
+    CreateAPIView,
+    ListAPIView,
+    RetrieveUpdateAPIView,
+    UpdateAPIView
+)
+from rest_framework.permissions import IsAuthenticated, IsAdminUser, AllowAny
+from rest_framework.views import APIView
+
+from core.models import User
+from core.serializers import (
+    ChangePasswordSerializer,
+    UserCreateSerializer,
+    UserDetailsSerializer,
+    UserUpdateSerializer
+)
+from core.tokens import AccountActivationTokenGenerator
+
+
+class UserList(ListAPIView):
+    """
+    Return a list of all the existing users.
+    """
+    permission_classes = (IsAuthenticated, IsAdminUser)
+    queryset = User.objects.all().order_by('id')
+    serializer_class = UserDetailsSerializer
+
+
+class UserDetailOrUpdateView(RetrieveUpdateAPIView):
+    """
+    retrieve:
+    Return current user details.
+
+    put:
+    Update the current user details with new one.
+    """
+    permission_classes = (IsAuthenticated,)
+    serializer_class = UserUpdateSerializer
+
+    def retrieve(self, request, *args, **kwargs):
+        serializer = UserDetailsSerializer(request.user)
+        return Response(serializer.data, status=status.HTTP_200_OK)
+
+
+class UserCreateView(CreateAPIView):
+    """
+    Create a new user instance.
+    """
+    permission_classes = (AllowAny,)
+    serializer_class = UserCreateSerializer
+
+    def post(self, request, *args, **kwargs):
+        serializer = self.get_serializer(data=request.data)
+        if serializer.is_valid():
+            user = serializer.save()
+            if user:
+                user.send_activation_email(request)
+                return Response({'data': 'send email'}, status=status.HTTP_201_CREATED)
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+
+class ActivateAccount(APIView):
+    """
+    get:
+    Activate the user account.
+    """
+    permission_classes = (AllowAny,)
+
+    def get(self, request, uidb64, token, *args, **kwargs):
+        uid = force_text(urlsafe_base64_decode(uidb64))
+        user = User.objects.filter(pk=uid).first()
```

Comment:
@shavaiz-hussain No, we cannot use get() with first(), as get() only return 1 user so it doesn't have first() function in it and if we use get and the attribute doesn't exist/match it'll raise `DoesNotExist` exception for which I've to add try catch block. To avoid that I used filter().